libyul/optimiser/FunctionCallFinder.h

```diff
@@ -0,0 +1,47 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * AST walker that finds all calls to a function of a given name.
+ */
+
+#pragma once
+
+#include <libyul/optimiser/ASTWalker.h>
+
+#include <vector>
+
+namespace solidity::yul
+{
+
+/**
+ * AST walker that finds all calls to a function of a given name.
+ *
+ * Prerequisite: Disambiguator
+ */
+class FunctionCallFinder: ASTModifier
```

Comment:
Why does this needs to be a `ASTModifier`? It is only reading, isn' it?

Reply:
Or is it because returning `vector<FunctionCall*>` so operating on the result could modify the AST?

Reply:
Yes, in order to return a non-const pointer.